app/hooks/useGroups.ts

```diff
@@ -0,0 +1,9 @@
+import type { SerializeFrom } from '@remix-run/node';
+import { useRouteLoaderData } from '@remix-run/react';
+import type { loader } from '~/root';
+
+export function useGroups() {
```

Comment:
Same comment as with the `useEvents` hook. This `useGroups` hook may not be needed if we're not using the group data in other places in our app.